src/decoder/plugins/SidplayDecoderPlugin.cxx

```diff
@@ -65,6 +66,38 @@ static unsigned default_songlength;
 
 static bool filter_setting;
 
+#ifdef HAVE_SIDPLAYFP
+static const unsigned romSize = 8192;
+static std::unique_ptr<uint8_t[]> kernal(new uint8_t[romSize]());
+static std::unique_ptr<uint8_t[]> basic(new uint8_t[romSize]());
+
+static void loadRom(std::unique_ptr<uint8_t[]> &dump, const Path romPath)
+{
+	try
+	{
+		FileReader romDump(romPath);
+		if (romDump.GetSize() != romSize)
+		{
+			FormatWarning(sidplay_domain,
+				"Corrupted rom dump %s - %u bytes expected",
+				romPath.c_str(), romSize);
+		}
+		else
+		{
+			romDump.Read(dump.get(), romSize);
```

Comment:
Forgot to check the `Read()` return value.

Reply:
Oops. 

Is the size check sufficient or should additional MD5/SHA2 checksums be implemented?